src/settingsholder.h

```diff
@@ -14,43 +14,13 @@ class SettingsHolder final : public QObject {
   Q_OBJECT
   Q_DISABLE_COPY_MOVE(SettingsHolder)
 
-  Q_PROPERTY(bool ipv6Enabled READ ipv6Enabled WRITE setIpv6Enabled NOTIFY
-                 ipv6EnabledChanged)
-  Q_PROPERTY(bool localNetworkAccess READ localNetworkAccess WRITE
-                 setLocalNetworkAccess NOTIFY localNetworkAccessChanged)
-  Q_PROPERTY(bool unsecuredNetworkAlert READ unsecuredNetworkAlert WRITE
-                 setUnsecuredNetworkAlert NOTIFY unsecuredNetworkAlertChanged)
-  Q_PROPERTY(bool captivePortalAlert READ captivePortalAlert WRITE
-                 setCaptivePortalAlert NOTIFY captivePortalAlertChanged)
-  Q_PROPERTY(bool startAtBoot READ startAtBoot WRITE setStartAtBoot NOTIFY
-                 startAtBootChanged)
-  Q_PROPERTY(bool protectSelectedApps READ protectSelectedApps WRITE
-                 setProtectSelectedApps NOTIFY protectSelectedAppsChanged)
-  Q_PROPERTY(bool gleanEnabled READ gleanEnabled WRITE setGleanEnabled NOTIFY
-                 gleanEnabledChanged)
-  Q_PROPERTY(
-      bool serverSwitchNotification READ serverSwitchNotification WRITE
-          setServerSwitchNotification NOTIFY serverSwitchNotificationChanged)
-  Q_PROPERTY(bool connectionChangeNotification READ connectionChangeNotification
-                 WRITE setConnectionChangeNotification NOTIFY
-                     connectionChangeNotificationChanged)
+#define SETTING(type, toType, getter, setter, ...) \
+  Q_PROPERTY(type getter READ getter WRITE setter NOTIFY settingsChanged)
+
```

Comment:
I'm a bit cautious to put all properties under one signal, as this would mean to re-evaluate all bindings that exist. Which is a lot of event overhead, IMHO.   
What if we would let the macro also define the old Signals that we had? e.g:

```suggestion
#define SETTING(type, toType, getter, setter, ...) \
  Q_SIGNAL void getter##Changed(); \
  Q_PROPERTY(type getter READ getter WRITE setter NOTIFY getter##Changed)

```

Reply:
I tried. This does not work because Q_SIGNAL is processed by `moc` and that tool does not support C++ macros. So, the signal will not be implemented in the `moc_settingholder.cpp`.

Reply:
If we get into the world of metaprogramming, we should be able to expose the settings as dynamic properties. That might enable us to cut down on the boilerplate macros, while still retaining the ability to access them via QML. I'm not sure if that allows us to dynamically create signals though.

Reply:
If we want to go way into the weeds of dynamic `Objects`, there might be some breadcrumbs here: https://doc.qt.io/archives/qq/qq16-dynamicqobject.html